oauth2/org.wso2.dpop.proof.generator/src/main/java/org/wso2/dpop/proof/generator/DPOPProofGenerator.java

```diff
@@ -0,0 +1,235 @@
+            if (!input.isEmpty()) {
+                return input;
+            }
+            log.warn("Input cannot be empty. Please try again.");
+        }
+    }
+    /**
+     * Generates a signed DPoP JWT using the provided key files.
+     *
+     * @param privateKeyPath Path to the private key file.
+     * @param publicKeyPath  Path to the public key file.
+     * @param keyPairType    Key type (EC or RSA).
+     * @param httpMethod     HTTP method (e.g., GET, POST).
+     * @param httpUrl        Target HTTP URL.
+     * @throws Exception If an error occurs during key loading or JWT signing.
+     */
+    private static void generateDPoPProof(String privateKeyPath, String publicKeyPath, String keyPairType,
+                                          String httpMethod, String httpUrl, String accessToken) throws Exception {
+
+        // Load keys
+        PrivateKey privateKey = loadPrivateKey(privateKeyPath, keyPairType);
+        PublicKey publicKey = loadPublicKey(publicKeyPath, keyPairType);
+
+        // Create JWK (JSON Web Key)
+        JWK jwk = createJWK(keyPairType, publicKey);
+
+        // Build JWT claims
+        JWTClaimsSet.Builder jwtClaimsSetBuilder = new JWTClaimsSet.Builder();
+        jwtClaimsSetBuilder.issuer("wso2_dpop_client")
+                .subject("sub")
+                .issueTime(new Date())
+                .jwtID(UUID.randomUUID().toString())
+                .notBeforeTime(new Date())
+                .claim("htm", httpMethod)
+                .claim("htu", httpUrl)
+                .expirationTime(new Date(System.currentTimeMillis() + 86400000L));
+        if (StringUtils.isNotBlank(accessToken)) {
+            log.info("access token: " + accessToken);
+            jwtClaimsSetBuilder.claim("ath", generateAccessTokenHash(accessToken));
+        }
+
+        JWTClaimsSet jwtClaimsSet = jwtClaimsSetBuilder.build();
+
+        // Create JWT Header
+        JWSHeader.Builder headerBuilder = ("EC".equals(keyPairType))
+                ? new JWSHeader.Builder(JWSAlgorithm.ES256)
+                : new JWSHeader.Builder(JWSAlgorithm.RS256);
+
+        headerBuilder.type(new JOSEObjectType("dpop+jwt"));
+        headerBuilder.jwk(jwk);
+
+        // Sign JWT
+        SignedJWT signedJWT = new SignedJWT(headerBuilder.build(), jwtClaimsSet);
+        signJWT(signedJWT, privateKey, keyPairType);
+
+        // Log results
+        log.info("[Signed JWT] : " + signedJWT.serialize());
+        log.info("[ThumbPrint] : " + jwk.computeThumbprint().toString());
+    }
+
+    /**
+     * Loads a private key from a file.
+     *
+     * @param keyPath    Path to the private key file.
+     * @param keyType    Key type (EC or RSA).
+     * @return PrivateKey object.
+     * @throws IOException, NoSuchAlgorithmException, InvalidKeySpecException
+     */
+    private static PrivateKey loadPrivateKey(String keyPath, String keyType)
+            throws IOException, NoSuchAlgorithmException, InvalidKeySpecException {
```

Comment:
Add a new line. Check in other places as well

Reply:
fixed with https://github.com/wso2/samples-is/pull/486/commits/340ccf50cbf8a8c486e9b43d2438f001348050a1